src/js/polyfills/blob.js

```diff
@@ -1,4 +1,240 @@
+     * of the blob consists of the concatenation of the values given
+     * in the parameter array.
+     *
+     * @param {*} blobParts
+     * @param {{ type?: string, endings?: string }} [options]
+     */
+    constructor (blobParts = [], options = {}) {
+        if (typeof blobParts !== 'object' || blobParts === null) {
+            throw new TypeError('Failed to construct \'Blob\': The provided value cannot be converted to a sequence.');
+        }
+
+        if (typeof blobParts[Symbol.iterator] !== 'function') {
+            throw new TypeError('Failed to construct \'Blob\': The object must have a callable @@iterator property.');
+        }
+
+        if (typeof options !== 'object' && typeof options !== 'function') {
+            throw new TypeError('Failed to construct \'Blob\': parameter 2 cannot convert to dictionary.');
+        }
+
+        if (options === null) {
+            options = {};
+        }
+
+
+        const encoder = new TextEncoder();
+
+        for (const element of blobParts) {
+            let part;
+
+            if (isView(element)) {
+                part = new Uint8Array(
+                    element.buffer.slice(
+                        element.byteOffset,
+                        element.byteOffset + element.byteLength
+                    )
+                );
+            } else if (element instanceof ArrayBuffer) {
+                part = new Uint8Array(element.slice(0));
+            } else if (element instanceof Blob) {
+                part = element;
+            } else {
+                part = encoder.encode(`${element}`);
+            }
+
+            const size = isView(part) ? part.byteLength : part.size;
+
+            // Avoid pushing empty parts into the array to better GC them
+            if (size) {
+                this.#size += size;
+                this.#parts.push(part);
+            }
+        }
+
+        this.#endings = `${options.endings === undefined ? 'transparent' : options.endings}`;
+        const type = options.type === undefined ? '' : String(options.type);
+
+        this.#type = /^[\x20-\x7E]*$/.test(type) ? type : '';
+    }
+
+    /**
+     * The Blob interface's size property returns the
+     * size of the Blob in bytes.
+     */
+    get size () {
+        return this.#size;
+    }
+
+    /**
+     * The type property of a Blob object returns the MIME type of the file.
+     */
+    get type () {
+        return this.#type;
+    }
+
+    /**
+     * The text() method in the Blob interface returns a Promise
+     * that resolves with a string containing the contents of
+     * the blob, interpreted as UTF-8.
+     *
+     * @return {Promise<string>}
+     */
+    async text () {
+        // More optimized than using this.arrayBuffer()
+        // that requires twice as much ram
+        const decoder = new TextDecoder();
+        let str = '';
+
+        for await (const part of this.stream()) {
+            str += decoder.decode(part, { stream: true });
```

Comment:
We don't currently support the stream parameter, and isn't it deprecated?

Reply:
um...? i'm confused. 
i did not find anything on [mdn](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder/decode#specifications), the [spec](https://encoding.spec.whatwg.org/#ref-for-dom-textdecoder-decode), or on [github](https://github.com/whatwg/encoding/issues) about it being deprecated. 

i also tried it out in txiki and streaming dose seem to be supported?

![image](https://github.com/saghul/txiki.js/assets/1148376/748042d2-9f0d-4087-8fc2-1a926becd4ba)

as to compared to:

```js
decoder.decode(new Uint8Array([226, 153]));
// "��"
decoder.decode(new Uint8Array([165]));
// "�"
```

Reply:
You are right, not sure what I confused this with 😅 It it supported indeed, in the polyfill we use. This is ready to go then!